app/auth/authV2/actions/RetrieveClientData.scala

```diff
@@ -47,23 +49,29 @@ class RetrieveClientData @Inject()(sessionDataService: SessionDataService,
 
     val useSessionDataService = appConfig.isSessionDataStorageEnabled
 
-    sessionDataService.getSessionData(useCookie = !useSessionDataService).map {
-      case Right(sessionData) => Right(ClientDataRequest(
-        sessionData.mtditid,
-        r.session.get(SessionKeys.clientFirstName),
-        r.session.get(SessionKeys.clientLastName),
-        sessionData.nino,
-        sessionData.utr,
-        getBooleanFromSession(SessionKeys.isSupportingAgent),
-        confirmed = {
-          r.session.get(SessionKeys.confirmedClient) match {
-            case Some(value) if value != "" => value.toBoolean
-            case _ => useSessionDataService
-          }
+    sessionDataService.getSessionData(useCookie = !useSessionDataService).flatMap {
+      case Right(sessionData) =>
+        clientDetailsService.checkClientDetails(sessionData.utr).map {
+          case Right(name) => Right(ClientDataRequest(
+              sessionData.mtditid,
+              name.firstName,
+              name.lastName,
+              sessionData.nino,
+              sessionData.utr,
+              getBooleanFromSession(SessionKeys.isSupportingAgent),
+              confirmed = {
+                r.session.get(SessionKeys.confirmedClient) match {
```

Comment:
my understanding is that we can fully rely on session-data service in this case, as client is confirmed right after we write to this service? not sure why we still need to have a split for this operation in two steps; might be something we need to discuss

Reply:
I'm not sure, are you saying we should just set confirmed to true in all cases? I'm still not sure what ConfirmedClient means to be honest

Reply:
we starting session/calling income-tax-session-data service in EnterClientsUTRController, but actual session should not start for the agent, until agent confirmed that its ok to continue for the client selected -> which is ConfirmClientUTRController: essentially starting a session for the agent is a two step operation;
 we did not discussed this originally and it seems that we are starting session for the agent in the wrong place;

Reply:
Ok, what would you like me to do in this PR?

Reply:
no, I've created a separate story for this: https://jira.tools.tax.service.gov.uk/browse/MISUV-8926